lightning/src/events/mod.rs

```diff
@@ -2438,3 +2449,19 @@ impl<T: EventHandler> EventHandler for Arc<T> {
 		self.deref().handle_event(event)
 	}
 }
+
+/// Wrapper time to move the bolt12 invoice and the static
+/// invoice across the same event as a unique type.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PaidInvoice {
+	/// Bolt12 invoice
+	Bolt12Invoice(crate::offers::invoice::Bolt12Invoice),
+	/// Static invoice
+	StaticInvoice(crate::offers::static_invoice::StaticInvoice),
+ }
+
+impl_writeable_tlv_based_enum_legacy!(PaidInvoice,
+	;
+	(0, Bolt12Invoice),
+	(2, StaticInvoice)
+);
```

Comment:
Ok after applying the https://github.com/lightningdevkit/rust-lightning/pull/3640 this code is no longer valid, I am not sure if the macros need to be upgradated with the new trait too or there is another macro that I am not able to see?

cc @valentinewallace

Reply:
Solved in [2354e2c](https://github.com/lightningdevkit/rust-lightning/pull/3593/commits/2354e2c0420b25e941f1c9683046276f35dd7eb7)